freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/instance/MetadataSetupService.java

```diff
@@ -65,7 +69,10 @@ public int saveInstanceMetaData(Stack stack, Iterable<CloudVmMetaDataStatus> clo
             instanceMetaDataEntry.setPrivateId(privateId);
             instanceMetaDataEntry.setStartDate(clock.getCurrentTimeMillis());
             instanceMetaDataEntry.setSubnetId(cloudInstance.getSubnetId());
-            instanceMetaDataEntry.setAvailabilityZone(cloudInstance.getAvailabilityZone());
+            if (instanceMetaDataEntry.getAvailabilityZone() == null) {
+                LOGGER.debug("Set Availability Zone since it is null");
+                instanceMetaDataEntry.setAvailabilityZone(cloudInstance.getAvailabilityZone());
+            }
```

Comment:
if the cloudinstance has availabilityzone information, shouldn't we always set the instancemetadata info to it?